docs/guides/frontend-redux.md

```diff
@@ -0,0 +1,63 @@
+# Frontend Data Architecture
+
+This post describes the architecture used throughout [frontend-scaffold](https://github.com/Code-4-Community/frontend-scaffold), Code4Community's shared frontend template used by all client projects. The scaffold is a React application created with Create React App using the TypeScript template. It uses Redux for application state management. 
```

Comment:
I still don't really get *why* redux and state management is needed. Why can't we just put all of this stuff in cookies or something?

Reply:
Cookies are a way of persisting data while Redux is an in-memory store - it isn't persisted between browser page refreshes, but because we're in an single-page app, this isn't a bad thing. 

Without Redux, data that we load into the application via asynchronous API calls would likely be stored in React component state. This can get complicated as we start to build large apps - we either keep all of the data in a high-level component's state and pass it down via props (prop drilling) or reload it each time we need it. With Redux, there is one common store that can be accessed anywhere in the application.